### PR TITLE
chore: screenshots task cleanup

### PR DESCRIPTION
**Pull request by DevVersion**
Attempt to make the screenshot task more readable and easier to understand. The screenshot task includes a lot of magic (due to the Firebase functions JWT solution) and we should try our best to make it as readable as possible.

**Note**: This also fixes that the screenshot results are uploaded outside of Gulp (causing gulp to continue with other tasks while this is running)

### CODE REVIEW

File: tools/gulp/tasks/screenshots.ts [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
-import {
-  openScreenshotsBucket,
-  connectFirebaseScreenshots} from '../util/firebase';
-import {isTravisMasterBuild} from '../util/travis-ci';
 
+// Firebase provides TypeScript definitions that are only accessible from specific namespaces.
+// This means that those types are really long and it's nearly impossible to write a function that
+// doesn't exceed the maximum columns. Import the types from the namespace so they are shorter.
+import Database = firebaseAdmin.database.Database;
+import DataSnapshot = firebaseAdmin.database.DataSnapshot;
```
Comment (jelbourn) [inline tools/gulp/tasks/screenshots.ts]: `FirebaseDatabase` and `FirebaseDataSnapshot`?

Comment (DevVersion) [inline tools/gulp/tasks/screenshots.ts]: Would work as well, but the main intention was to keep it as short as possible.

Comment (jelbourn) [inline tools/gulp/tasks/screenshots.ts]: I aim for explicitness first and brevity second.

Comment (DevVersion) [inline tools/gulp/tasks/screenshots.ts]: Hmm not sure. It should be pretty clear from the file name and from the comment as well.

Comment (jelbourn) [inline tools/gulp/tasks/screenshots.ts]: I don'r care that much for this one

**Pull request merged**